### PR TITLE
Finish WebFlux transaction before popping scope

### PR DESCRIPTION
**Pull request by adinauer**
## :scroll: Description

Merge `SentryWebTracingFilter` into `SentryWebFilter` to avoid ordering problems with `popScope` and `transaction.finish`.

## :bulb: Motivation and Context

Fixes https://github.com/getsentry/sentry-java/issues/2700

## :pencil: Checklist

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryWebFilterWithThreadLocalAccessor.java [hunk 1]
```diff
@@ -21,17 +22,26 @@ public SentryWebFilterWithThreadLocalAccessor(final @NotNull IHub hub) {
...
             .filter(serverWebExchange)
             .doFinally(
-                __ -> {
-                  doFinally(Sentry.getCurrentHub());
-                  Sentry.setCurrentHub(NoOpHub.getInstance());
-                })
+                __ ->
+                    doFinally(
+                        serverWebExchange,
+                        Sentry.getCurrentHub(),
```
Comment (lbloder) [inline sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryWebFilterWithThreadLocalAccessor.java:33]: Does `Sentry.getCurrentHub()` work correctly in this filter method? `ReactorUtils.withSentryNewMainHubClone` does not set the current hub, so shouldn't we get the hub from the context?
Or is this somehow handled by reactor behind the scenes?

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryWebFilterWithThreadLocalAccessor.java [hunk 2]
```diff
@@ -21,17 +22,26 @@ public SentryWebFilterWithThreadLocalAccessor(final @NotNull IHub hub) {
...
+                        transactionContainer.transaction))
+            .doOnError(e -> doOnError(transactionContainer.transaction, e))
             .doFirst(
                 () -> {
                   doFirst(serverWebExchange, Sentry.getCurrentHub());
```
Comment (lbloder) [inline sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryWebFilterWithThreadLocalAccessor.java:38]: See comment about `Sentry.getCurrentHub()`

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryWebFilterWithThreadLocalAccessor.java [hunk 3]
```diff
@@ -21,17 +22,26 @@ public SentryWebFilterWithThreadLocalAccessor(final @NotNull IHub hub) {
...
+                  transactionContainer.transaction =
+                      maybeStartTransaction(Sentry.getCurrentHub(), serverWebExchange.getRequest());
```
Comment (lbloder) [inline sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryWebFilterWithThreadLocalAccessor.java:40]: See comment about `Sentry.getCurrentHub()`

**Review by lbloder — COMMENTED** [review]
Besides the questions about getting the Hub in `SentryWebFilterWithThreadLocalAccessor`, LGTM

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryWebFilterWithThreadLocalAccessor.java (hunk 1 above)
Comment (adinauer) [reply sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryWebFilterWithThreadLocalAccessor.java:33]: `ThreadLocal` is set via `ThreadLocalAccessor` and propagated via reactor context.

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryWebFilterWithThreadLocalAccessor.java (hunk 2 above)
Comment (adinauer) [reply sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryWebFilterWithThreadLocalAccessor.java:38]: same as above

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryWebFilterWithThreadLocalAccessor.java (hunk 3 above)
Comment (adinauer) [reply sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryWebFilterWithThreadLocalAccessor.java:40]: same as above

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryWebFilterWithThreadLocalAccessor.java (hunk 1 above)
Comment (adinauer) [reply sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryWebFilterWithThreadLocalAccessor.java:33]: Just tested again with debugger, works as expected (i.e. uses the correct hub).

**Review by cleptric — APPROVED** [review]

**Pull request merged**